zapcore/error.go

```diff
@@ -42,13 +43,29 @@ import (
 //      ...
 //    ],
 //  }
-func encodeError(key string, err error, enc ObjectEncoder) error {
+func encodeError(key string, err error, enc ObjectEncoder) (retErr error) {
+	// Try to capture panics (from nil references or otherwise) when calling
+	// the Error() method
+	defer func() {
+		if rerr := recover(); rerr != nil {
+			// If it's a nil pointer, just say "<nil>". The likeliest causes are a
+			// Stringer that fails to guard against nil or a nil pointer for a
+			// value receiver, and in either case, "<nil>" is a nice result.
+			if v := reflect.ValueOf(err); v.Kind() == reflect.Ptr && v.IsNil() {
+				enc.AddString(key, "<nil>")
+				return
+			}
+
+			retErr = fmt.Errorf("PANIC=%v", rerr)
+		}
+	}()
+
 	basic := err.Error()
 	enc.AddString(key, basic)
 
 	switch e := err.(type) {
 	case errorGroup:
-		return enc.AddArray(key+"Causes", errArray(e.Errors()))
+		retErr = enc.AddArray(key+"Causes", errArray(e.Errors()))
```

Comment:
we don't need to do this. the defer and the named return will still capture the value.

```suggestion
		return enc.AddArray(key+"Causes", errArray(e.Errors()))
```